ocean_provider/utils/datatoken.py

```diff
@@ -111,10 +112,31 @@ def verify_order_tx(
             f"Provider was not able to check the signed message in ProviderFees event\n"
         )
 
-    # TO DO - check transfer of providerFeeAmount providerFeeToken to providerFeeAddress
-
+    # check duration
+    if provider_fee_order_log.args.duration > 0:
+        ts = datetime.now().timestamp()
```

Comment:
I think you can call this variable `now` or something more relevant, since "ts" can be any timestamp.